glusterd2/commands/peers/addpeer.go

```diff
@@ -27,6 +31,11 @@ func addPeerHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if peer.GetMetadataLen(req.Metadata) > maxMetadataSizeLimit {
```

Comment:
You have to check this:

(current metadata size + metadata size from request) > maxMetadataSizeLimit

Reply:
As discussed, current metadata size + metadata size from request for addpeer won't make any sense since add peer current metadata size will be 0.